### ISSUE TITLE
Test failure with sphinx 1.5.1

### ISSUE THREAD

**Issue by jamesjer** [mcmtroffaes/sphinxcontrib-bibtex#111]
The 0.3.4 build is currently broken for the Fedora Linux distribution, using sphinx 1.5.1, because of a failed test:

FAIL: test_latex_refs.test_latex_refs
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/sphinx_testing/util.py", line 142, in decorator
    func(*(args + (app, status, warning)), **kwargs)
  File "/builddir/build/BUILD/python-sphinxcontrib-bibtex-0.3.4/sphinxcontrib-bibtex-0.3.4/test/test_latex_refs.py", line 23, in test_latex_refs
    assert re.search('\\hyperref\[contents:huygens\]', output)

Changing these lines in test/test_latex_refs.py:

    assert re.search('\\hyperref\[contents:huygens\]', output)
    assert re.search('\\label{contents:huygens}', output)

to this:

    assert re.search('\\hyperref\[\\\\detokenize{contents:huygens}\]', output)
    assert re.search('\\label{\\\\detokenize{contents:huygens}}', output)

fixes the issue, but is wrong for earlier versions of sphinx.

**Comment by mcmtroffaes** [comment]
I've pushed a new release; the test should now pass. Thanks for reporting!

**Comment by jamesjer** [comment]
Thanks for the fix, but now the build doesn't work for sphinx < 1.5.1.  It looks like there is an extra '}' in there:

  '\\hyperref\[(\\\\detokenize{)?contents:huygens(})?}\]', output)

There is an optional parenthesized '}', which is correct, but then there is another '}' right after the '?'.

**Comment by mcmtroffaes** [comment]
Oh gosh... Hang on.

**Comment by mcmtroffaes** [comment]
Should be good now; can you confirm? I'll roll another release then.

**Comment by ghisvail** [comment]
Still getting a failure on Debian with Sphinx 1.5.6:
```
======================================================================
ERROR: test_latex_refs.test_latex_refs
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/usr/lib/python3/dist-packages/sphinx_testing/util.py", line 142, in decorator
    func(*(args + (app, status, warning)), **kwargs)
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_3.6/build/test/test_latex_refs.py", line 24, in test_latex_refs
    '\\hyperref\[(\\\\detokenize{)?contents:huygens(})?\]', output)
  File "/usr/lib/python3.6/re.py", line 182, in search
    return _compile(pattern, flags).search(string)
  File "/usr/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.6/sre_parse.py", line 855, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File "/usr/lib/python3.6/sre_parse.py", line 502, in _parse
    code = _escape(source, this, state)
  File "/usr/lib/python3.6/sre_parse.py", line 401, in _escape
    raise source.error("bad escape %s" % escape, len(escape))
sre_constants.error: bad escape \h at position 0

----------------------------------------------------------------------
```
```

**Comment by ghisvail** [comment]
And for the other regex:
```
======================================================================
ERROR: test_latex_refs.test_latex_refs
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/usr/lib/python3/dist-packages/sphinx_testing/util.py", line 142, in decorator
    func(*(args + (app, status, warning)), **kwargs)
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_3.6/build/test/test_latex_refs.py", line 26, in test_latex_refs
    '\\label{(\\\\detokenize{)?contents:huygens(})?}}', output)
  File "/usr/lib/python3.6/re.py", line 182, in search
    return _compile(pattern, flags).search(string)
  File "/usr/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.6/sre_parse.py", line 855, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File "/usr/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File "/usr/lib/python3.6/sre_parse.py", line 502, in _parse
    code = _escape(source, this, state)
  File "/usr/lib/python3.6/sre_parse.py", line 401, in _escape
    raise source.error("bad escape %s" % escape, len(escape))
sre_constants.error: bad escape \l at position 0

----------------------------------------------------------------------
```

**Comment by mcmtroffaes** [comment]
Fixed now in develop (was related to Python 3.6).

**Comment by ghisvail** [comment]
@mcmtroffaes Thanks for fixing the issues I recently reported. Would you consider uploading a new release please? I would like to use it for the initial packaging for Debian, instead of carrying a large patch queue on top of an old release.

Thanks,
